cms/static/js/views/metadata.js

```diff
@@ -220,6 +225,44 @@ function(BaseView, _, MetadataModel, AbstractEditor, FileUpload, UploadDialog, V
 
     });
 
+    Metadata.NumberWithWarning = Metadata.Number.extend({
+        initialize: function() {
+            Metadata.Number.prototype.initialize.apply(this);
+            var options = this.model.get("options");
+            this.msg = options["warning"];
+            // show initial warning if necessary
+            this.checkWarning();
+        },
+
+        changed: function() {
+            Metadata.Number.prototype.changed.apply(this);
+            this.checkWarning();
+        },
+
+        checkWarning: function() {
+            var value = this.getValueFromEditor();
+            if (parseInt(value)) {
+                this.showWarning();
+            } else {
+                this.removeWarning(); 
+            }
+        },
+
+        showWarning: function() {
+            if (this.$(".setting-help-warning").length < 1) {
+                var $warning = this.$(".setting-help").clone();
+                $warning
```

Comment:
If we're going to use chaining, we could remove this line and join on the initial declaration. Otherwise, let's be explicit and call `$warning.WHATEVER` each time.